pkg/resource/events-logger-config/grafana-agent-config.go

```diff
@@ -60,39 +44,37 @@ func generateGrafanaAgentConfig(lc loggedcluster.Interface, defaultWorkloadClust
 		GrafanaAgentInnerConfig: grafanaAgentInnerConfig,
 	}
 
-	err = grafanaAgentTemplate.Execute(&values, data)
+	err = grafanaAgentConfigTemplate.Execute(&values, data)
 	if err != nil {
 		return "", err
 	}
 
 	return values.String(), nil
 }
 
-func generateGrafanaAgentInnerConfig(lc loggedcluster.Interface, defaultWorkloadClusterNamespaces []string) (string, error) {
+func generateGrafanaAgentInnerConfig(lc loggedcluster.Interface, defaultNamespaces []string) (string, error) {
 	var values bytes.Buffer
 
 	data := struct {
 		ClusterID          string
 		Installation       string
 		InsecureSkipVerify string
 		SecretName         string
+		SecretNamespace    string
 		ScrapedNamespaces  string
 	}{
 		ClusterID:          lc.GetClusterName(),
 		Installation:       lc.GetInstallationName(),
 		InsecureSkipVerify: fmt.Sprintf("%t", lc.IsInsecureCA()),
-		SecretName:         fmt.Sprintf("%s-%s", lc.GetClusterName(), common.GrafanaAgentExtraSecretName()),
-		ScrapedNamespaces:  common.FormatScrapedNamespaces(lc, defaultWorkloadClusterNamespaces),
+		SecretName:         eventsloggersecret.GetEventsLoggerSecretName(lc),
```

Comment:
Remove this secret and enforces kube-system in thé template

Reply:
We already have `kube-system` hardcoded in the grafana-agent template so I can indeed remove the `SecretNamespace` field.

As for the `SecretName` one we still need to keep it defined as it is right now as the name will depend on the WC's right ?

Reply:
Yes but i'm reviewing on m'y phone so I might have lost context 😅